src/dragonfs.c

```diff
@@ -485,7 +485,7 @@ static directory_entry_t *find_dirent(char *name, directory_entry_t *cur_node)
 static int recurse_path(const char * const path, int mode, directory_entry_t **dirent, int type)
 {
     int ret = DFS_ESUCCESS;
-    char token[MAX_FILENAME_LEN+1];
+    char token[MAX_FILENAME_LEN+1] = {0};
```

Comment:
Can you please just write `token[0] = 0;` at the beginning of the function? The syntax that you used forces a full zero-initialization of a 243 array on the stack, which needlessly pollutes 16 data cachelines.

Reply:
Fair enough. I've pushed the modifications. Let me know if you want me to rebase and squash the commits.